homeassistant/components/intellifire/sensor.py

```diff
@@ -0,0 +1,125 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import datetime
+
+from intellifire4py import IntellifirePollData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN
+
+
+@dataclass
+class IntellifireSensorRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    # Although sensors could have a variety of different return values,
+    # all the ones below are only returning ints
+    value_fn: Callable[[IntellifirePollData], int | str]
+
+
+@dataclass
+class IntellifireSensorEntityDescription(
+    SensorEntityDescription, IntellifireSensorRequiredKeysMixin
+):
+    """Describes a sensor sensor entity."""
+
+
+INTELLIFIRE_SENSORS: tuple[IntellifireSensorEntityDescription, ...] = (
+    IntellifireSensorEntityDescription(
+        key="flame_height",
+        icon="mdi:fire-circle",
+        name="Flame Height",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.flameheight,
+    ),
+    IntellifireSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda data: data.temperature_c,
+    ),
+    IntellifireSensorEntityDescription(
+        key="target_temp",
+        name="Target Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda data: data.thermostat_setpoint_c,
+    ),
+    IntellifireSensorEntityDescription(
+        key="fan_speed",
+        icon="mdi:fan",
+        name="Fan Speed",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.fanspeed,
+    ),
+    IntellifireSensorEntityDescription(
+        key="timer_end_time",
+        icon="mdi:timer-sand",
+        name="Timer End",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: "Stopped"
+        if (data.timeremaining_s == 0)
+        else (
+            datetime.datetime.now() + datetime.timedelta(seconds=data.timeremaining_s)
+        ).strftime("%X"),
```

Comment:
`Stopped` is not a valid value to return in this case.

Return a Python datetime object and set the device class to `SensorDeviceClass.TIMESTAMP` (so the frontend knows how to handle it).

Reply:
Will do!

If off it will return `None`
<img width="309" alt="image" src="https://user-images.githubusercontent.com/6491743/151161847-a6ba51cd-595a-4971-8b39-1cd9d27c7975.png">

If On it will return

<img width="311" alt="image" src="https://user-images.githubusercontent.com/6491743/151161915-20479a9b-e625-4f14-adc8-ffc415dcfdb3.png">

As per documentation returns `None` or `datetime`

Reply:
I wonder if it makes sense to show the miliseconds...